File: multiples_of_3_and_5.rb
# If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

# Find the sum of all the multiples of 3 or 5 below 1000.

def multiples_of_3_or_5()
  index = 1
  numbers = []
  while index < 1000
    if index % 3 == 0 || index % 5 == 0
      numbers << index
    end
    index += 1
  end
  return numbers.sum
end

p multiples_of_3_or_5

File: c++/multiples_of_3_and_5.cpp
/*
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
Find the sum of all the multiples of 3 or 5 below 1000.
*/

#include <iostream>
#include <vector>
using namespace std;

int main()
{
  vector<int> numbers;
  for (int i = 1; i < 1000; i++)
  {
    if (i % 3 == 0 || i % 5 == 0)
    {
      numbers.push_back(i);
    }
  }
  int total = 0;
  for (int i = 0; i < numbers.size(); i++)
  {
    total += numbers[i];
  }
  cout << total << endl;
  return 0;
}